site/app/controllers/admin/GradeOverrideController.php

```diff
@@ -40,8 +40,42 @@ public function getOverriddenGrades($gradeable_id) {
 
     #[Route("/courses/{_semester}/{_course}/grade_override/{gradeable_id}/delete", methods: ["POST"])]
     public function deleteOverriddenGrades($gradeable_id) {
-        $this->core->getQueries()->deleteOverriddenGrades($_POST['user_id'], $gradeable_id);
-        return $this->getOverriddenGrades($gradeable_id);
+        $team = $this->core->getQueries()->getTeamByGradeableAndUser($gradeable_id, $_POST['user_id']);
+        //0 is for single submission, 1 is for team submission
+        $option = isset($_POST['option']) ? $_POST['option'] : -1;
+        if ($team != null && $team->getSize() > 1) {
+            if ($option == 0) {
+                $this->core->getQueries()->deleteOverriddenGrades($_POST['user_id'], $gradeable_id);
+                return $this->getOverriddenGrades($gradeable_id);
+            }
+            elseif ($option == 1) {
+                $team_member_ids = explode(", ", $team->getMemberList());
+                for ($i = 0; $i < count($team_member_ids); $i++) {
+                    $this->core->getQueries()->deleteOverriddenGrades($team_member_ids[$i], $gradeable_id);
+                }
+                return $this->getOverriddenGrades($gradeable_id);
+            }
+            else {
+                $team_member_ids = explode(", ", $team->getMemberList());
+                $team_members = [];
+                for ($i = 0; $i < count($team_member_ids); $i++) {
+                    $team_members[$team_member_ids[$i]] = $this->core->getQueries()->getUserById($team_member_ids[$i])->getDisplayedGivenName() . " " .
+                        $this->core->getQueries()->getUserById($team_member_ids[$i])->getDisplayedFamilyName();
+                }
+                $popup_html = $this->core->getOutput()->renderTwigTemplate(
+                    "admin/users/MoreGradeOverrides.twig",
+                    ['member_list' => $team_members, 'is_delete' => true]
+                );
+                return $this->core->getOutput()->renderJsonSuccess([
+                    'is_team' => true,
+                    'popup' => $popup_html
+                ]);
```

Comment:
Sending HTML as a field in a JSON response seems like a poor idea.  A better option would be to send the data via this response, and then deal with rendering HTML on the client side.

Reply:
Can JS render a twig template? I don't see an example of this in the code.

Reply:
One possible approach is to render a Twig template and then use JS to manipulate the data within it.  You'd render a Twig template with the general structure of the HTML, and then use jQuery to fill it with data.